cmd/posix-errors.go

```diff
@@ -69,8 +65,7 @@ func isSysErrIO(err error) bool {
 // Check if the given error corresponds to EISDIR (is a directory).
 func isSysErrIsDir(err error) bool {
 	if pathErr, ok := err.(*os.PathError); ok {
-		switch pathErr.Err {
-		case syscall.EISDIR:
+		if pathErr.Err == syscall.EISDIR {
```

Comment:
Further simplification is possible simply do `return pathErr.Err == syscall.EISDIR`